src/NonEmptyArray2v.ts

```diff
@@ -33,7 +33,6 @@ export interface NonEmptyArray<A> extends Array<A> {
   0: A
   map: <B>(f: (a: A, index: number, nea: NonEmptyArray<A>) => B) => NonEmptyArray<B>
   concat: (as: Array<A>) => NonEmptyArray<A>
-  reverse: () => NonEmptyArray<A>
```

Comment:
What about extending `ReadonlyArray<A>`? This will drop all mutable methods.

Reply:
@raveclassic we can't:

- raises a compiler error (`Interface 'NonEmptyArray<A>' incorrectly extends interface 'readonly A[]'...`)
- would be a breaking change

Also handling `ReadonlyArray`s is just annoying without providing real benefits over a lint rule